pkg/k8s/apps/interface.go

```diff
@@ -19,21 +19,16 @@ import (
 	"github.com/okteto/okteto/pkg/model"
 	appsv1 "k8s.io/api/apps/v1"
 	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 )
 
 type App interface {
-	Name() string
-	Kind() string
+	TypeMeta() metav1.TypeMeta
+	ObjectMeta() metav1.ObjectMeta
+	TemplateObjectMeta() metav1.ObjectMeta
+
 	Replicas() int32
-	GetLabel(key string) string
```

Comment:
I think it's a good improvement. I like that the interface is smaller and we rely on the Meta abstraction which is stronger due to it being part of all the underlying types we will probably use or easily implementable .